dotnet/src/Connectors/Connectors.Postgres.UnitTests/PostgresVectorStoreCollectionSqlBuilderTests.cs

```diff
@@ -0,0 +1,418 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.VectorData;
+using Microsoft.SemanticKernel.Connectors.Postgres;
+using Pgvector;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SemanticKernel.Connectors.Postgres.UnitTests;
+
+public class PostgresVectorStoreCollectionSqlBuilderTests
+{
+    private readonly ITestOutputHelper _output;
+    private static readonly float[] s_vector = new float[] { 1.0f, 2.0f, 3.0f };
+
+    public PostgresVectorStoreCollectionSqlBuilderTests(ITestOutputHelper output)
+    {
+        this._output = output;
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void TestBuildCreateTableCommand(bool ifNotExists)
+    {
+        var builder = new PostgresVectorStoreCollectionSqlBuilder();
+
+        var recordDefinition = new VectorStoreRecordDefinition()
+        {
+            Properties = [
+                new VectorStoreRecordKeyProperty("id", typeof(long)),
+                new VectorStoreRecordDataProperty("name", typeof(string)),
+                new VectorStoreRecordDataProperty("code", typeof(int)),
+                new VectorStoreRecordDataProperty("rating", typeof(float?)),
+                new VectorStoreRecordDataProperty("description", typeof(string)),
+                new VectorStoreRecordDataProperty("parking_is_included", typeof(bool)),
```

Comment:
Might be worth having a record definition where some of the properties have storage names that are different to the main property name, to make sure these are used correctly in the command.